pkg/upgrade/migrate.go

```diff
@@ -160,27 +161,29 @@ func Migrate(ctxt context.Context, c client.Interface, nodename string) error {
 		return fmt.Errorf("failed to close host-local IPAM data dir directory: %s", err)
 	}
 
-	// Disable cni by setting datastoreReady to false.
+	// Disable cni by setting DatastoreReady to false.
 	log.Info("setting datastore readiness to false")
-	clusterInfo, err := c.ClusterInformation().Get(ctxt, "default", options.GetOptions{})
-	if err != nil {
-		return fmt.Errorf("failed to fetch cluster information: %s", err)
-	}
-	if *clusterInfo.Spec.DatastoreReady {
-		*clusterInfo.Spec.DatastoreReady = false
-		for i := uint(0); i < 5; i++ {
+	var clusterInfo *v3.ClusterInformation
+	for i := uint(0); i < 5; i++ {
+		clusterInfo, err = c.ClusterInformation().Get(ctxt, "default", options.GetOptions{})
+		if err != nil {
+			return fmt.Errorf("failed to fetch cluster information: %s", err)
+		}
+		if clusterInfo.Spec.DatastoreReady == nil || *clusterInfo.Spec.DatastoreReady {
+			f := false
+			clusterInfo.Spec.DatastoreReady = &f
 			if clusterInfo, err = c.ClusterInformation().Update(ctxt, clusterInfo, options.SetOptions{}); err != nil {
 				if _, ok := err.(errors.ErrorResourceUpdateConflict); ok {
 					log.Info("Encountered update conflict, retrying...")
-					time.Sleep((1 << i) * time.Second)
+					time.Sleep(1 * time.Second)
```

Comment:
no more exponential backoff?

Reply:
Nope, I don't think it's what we want here. 

This shouldn't fail, and if it does it shouldn't be persistent - the exponential backoff will just make this take longer than it should.